libraries/logging/stream-logger/src/main/java/io/getstream/logging/StreamLogger.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.logging
+
+public typealias Priority = Int
+
+/**
+ * Low-level Logger interface.
+ */
+public interface StreamLogger {
+
+    public companion object Level {
+        /** Priority constant for the [log] method; use [StreamLog.v]. */
+        public const val VERBOSE: Priority = 2
+        /** Priority constant for the [log] method; use [StreamLog.d]. */
+        public const val DEBUG: Priority = 3
+        /** Priority constant for the [log] method; use [StreamLog.i]. */
+        public const val INFO: Priority = 4
+        /** Priority constant for the [log] method; use [StreamLog.w]. */
+        public const val WARN: Priority = 5
+        /** Priority constant for the [log] method; use [StreamLog.e]. */
+        public const val ERROR: Priority = 6
+        /** Priority constant for the [log] method. */
+        public const val ASSERT: Priority = 7
+    }
```

Comment:
What about to have an enum class for those type?

Reply:
I keep as `Int` to make logic like
```kotlin
val validator = IsLoggableValidator { priority: Int, tag: String ->  
    priority > StreamLogger.INFO && tag in validTags
}
```

Reply:
You can have almost the same logic with enum:
```
enum class Priority(val level: Int) {
    VERBOSE(2),
    DEBUG(3),
    INFO(4),
    WARN(5),
    ERROR(6),
    ASSERT(7),
}

val validator = IsLoggableValidator { priority: Priority, tag: String ->
    priority.level > Priority.INFO.level
}
```
We can create a function into our enum class to return if the level is bigger

Reply:
Using a "real type", other devs can know which values they can use without digging into the class to see which values are valid and which one not, allowing us to refactor it in the future without breaking our customers logic

Reply:
What is a big advantage of turing it into `enum`?
I won't use enum as it is cause I still will be converting names to:
```kotlin
 VERBOSE -> "V"
```

Reply:
What is the advantage of using `typealias Priority = Int`? 
It does nothing because we don't limit what you can use to define the log level.
Keeping a good type structure on our code helps us to reduce bugs in the future.
If you use Int, and for some reason, we need to introduce an intermediate value, it will break all our customer's apps.
Instead of that, if we have a clear data structure (In this case, an enum type), it won't break our customer's apps because they would be using "an Int value" that doesn't make sense